### PR TITLE
add support for ebooks

### LINKED ISSUE #60: [Feature request] book format conversion
I like the morphos convertes instance. It has really lot possibility. 
However there is one category what I personally use frequently, but morphos doesn't support it yet.

Different book style conversation.

I aware of the following book formats:

- doc
- docx
- epub
- fb2
- mobi
- txt
- pdf

Probably there are more but I have experience with these.
Normally what I want to do as convertion to grab a file from the previous list and create a epub version.

Another usecase, when I have even a pdf or just a bunch of jpeg and I want to make a e-comic with cbz or cbr format.

I hope this type of usage would be fit for the morphos project.
Thanks in advance.

### PR DESCRIPTION
**Pull request by danvergara**
# Add support for ebooks

## Description

This PR adds support for some of the most common e-books. To do this I had to embed the [ebook-convert](https://manual.calibre-ebook.com/generated/en/ebook-convert.html) binary from the Calibre project. The tool is installed on the Container image.

Fixes #60 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Added more tests and QA'd the new conversions.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: pkg/files/documents/pdf.go
Comment (Sirpyerre) [inline pkg/files/documents/pdf.go:1]: Veo muy largo el archivo, quizá valga la pena separarse por responsabilidades o funciones.

File: pkg/files/documents/pdf.go [hunk 1]
```diff
@@ -383,19 +384,100 @@ func (p *Pdf) ConvertTo(fileType, subType string, file io.Reader) (io.Reader, er
...
 
 			// Reads the zip file as an slice of bytes.
-			zipFile, err := os.ReadFile(zipFileName)
+			zipFile, err := os.ReadFile(archive.Name())
 			if err != nil {
 				return nil, fmt.Errorf("error reading zip file: %v", err)
 			}
 
 			return bytes.NewReader(zipFile), nil
+		case EPUB:
```
Comment (Sirpyerre) [inline pkg/files/documents/pdf.go:393]: El case para EPUB es muy largo, ¿por qué no lo metes en un método aparte el cuerpo de ese case?

Comment (ThiagoMowszet) [reply pkg/files/documents/pdf.go:393]: Coincido con Sirpyerre aqui. Lo haria en un metodo aparte.

**Review by ThiagoMowszet — APPROVED** [review]

**Review by ThiagoMowszet — APPROVED** [review]

**Comment by danvergara** [comment]
@ThiagoMowszet @Sirpyerre  I'm not done with this one, guys. I appreciate the early review, but I'm far from done adding support for ebooks. When the time to properly review this, I'll move the PR from draft mode to ready for review.

A refactor is dramatically needed but I'm more focused on move things along. When I reach a desired stated, I'll go over the code and build a solution off the most common patterns.

I'll let you know, guys.

**Comment by ThiagoMowszet** [comment]
> @ThiagoMowszet @Sirpyerre I'm not done with this one, guys. I appreciate the early review, but I'm far from done adding support for ebooks. When the time to properly review this, I'll move the PR from draft mode to ready for review.
> 
> A refactor is dramatically needed but I'm more focused on move things along. When I reach a desired stated, I'll go over the code and build a solution off the most common patterns.
> 
> I'll let you know, guys.

Thanks for the update @danvergara!

**Pull request merged**